File: linkedList/cppRealizations/mergeKSortedLists.cpp
#include <iostream>
#include <vector>

using namespace std;


struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};


class Solution {
public:
    ListNode* mergeKLists(vector<ListNode*>& lists) {
        
    }
};


int main() {
    vector<ListNode*> input = {new ListNode(1, new ListNode(4, new ListNode(5))), new ListNode(1, new ListNode(3, new ListNode(4))), new ListNode(2, new ListNode(6))};
    ListNode* out = Solution().mergeKLists(input);
    while(out){
        cout << out->val << endl;
        out = out->next;
    }
    return 0;
}

File: linkedList/mergeKSortedLists.py
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution(object):
    def mergeKLists(self, lists):
        pass


if __name__ == "__main__":
    input = [ListNode(1, ListNode(4, ListNode(5))), ListNode(1, ListNode(3, ListNode(4))), ListNode(2, ListNode(6))]
    out = Solution().mergeKLists(input)
    while out:
        print(out.val)
        out = out.next